db.js

```diff
@@ -194,6 +195,62 @@ exports.init = function (sbot, config) {
     )
   }
 
+  function addTransaction(msgVals, oooMsgVals, cb) {
+    const guard = guardAgainstDuplicateLogs('addTransaction()')
+    if (guard) return cb(guard)
+
+    oooMsgVals = oooMsgVals || []
+    msgVals = msgVals || []
+
+    onceWhen(
+      stateFeedsReady,
+      (ready) => ready === true,
+      () => {
+        function add(msgKeys, oooKeys) {
+          log.addTransaction(
+            msgKeys.concat(oooKeys),
+            msgVals.concat(oooMsgVals),
+            (err, kvts) => {
+              if (err) return cb(err)
+
+              kvts.forEach((kvt) => post.set(kvt))
+              cb(null, kvts)
+            }
+          )
+        }
+
+        validate2.validateOOOBatch(hmacKey, oooMsgVals, (err, oooKeys) => {
```

Comment:
Took me a while to understand this function, it's a bit funky with the separate keys, and the two different validations. I wonder if it would be simpler to use `multicb` to run both validateOOOBatch and validateBatch in parallel?

Reply:
Yeah, that refactor found a small bug (if validateOOO fails it was not caught) ;)